src/features/widgets/store/lib/spotPrice/fetchSpotPrices.tsx

```diff
@@ -0,0 +1,73 @@
+import { isEmpty } from 'lodash';
+
+export interface NativeTokenBalanceResponse {
+  data: {
+    base: string;
+    currency: string;
+    amount: string;
+  };
+}
+export type Payload = Record<string, { currency: string; price: string }>;
+export type PayloadPromise = Promise<Payload>;
+
+/**
+ * Get native token prices for selected currencies
+ * @param {Array<[string, string]>} pairIds - array of token pairs
+ * @returns Record with spot prices
+ */
+export const getSpotPrices = async (
+  pairIds: Array<[string, string]>
+): PayloadPromise => {
+  // Filter out XTZ as we get balance from different api
+  const tokenIds: Array<[string, string]> = pairIds.filter(
+    (pairId) => pairId[0] !== 'XTZ'
+  );
+  const nativeTokenIds: Array<[string, string]> = pairIds.filter(
+    (pairId) => pairId[1] !== 'XTZ'
+  );
+  let payload: Payload = {};
+
+  // 0 is XTZ id
+  if (nativeTokenIds) {
+    await Promise.all(
+      nativeTokenIds.map(async (pair: [string, string]): PayloadPromise => {
+        return await getNativeTokenBalance(pair);
```

Comment:
We've discussed this before with another feature, but awaiting a promise while constructing an array for Promise.all, makes the requests sequential.

But we will always have one native token spot price api call, so why deal in arrays in the first place?